rewrite-gradle/src/main/java/org/openrewrite/gradle/util/GradleWrapper.java

```diff
@@ -69,50 +71,125 @@ public class GradleWrapper {
+                    distributionType,
+                    null,
+                    null
+            );
         }
 
-        return new GradleVersion(version,
-                GRADLE_DISTRIBUTIONS_URL + "/gradle-" + version + "-" + distributionType.getFileSuffix() +".zip",
-                GRADLE_DISTRIBUTIONS_URL + "/gradle-" + version + "-" + distributionType.getFileSuffix() +".zip.sha256",
-                GRADLE_DISTRIBUTIONS_URL + "/gradle-" + version + "-wrapper.jar.sha256"
-        );
+        throw new IllegalStateException("Unsupported distribution url from Gradle wrapper version detection: " + currentDistributionUrl);
     }
 
-    public static List<GradleVersion> listAllVersions(ExecutionContext ctx) {
+    public static List<GradleVersion> listAllPublicVersions(ExecutionContext ctx) {
         HttpSender httpSender = HttpSenderExecutionContextView.view(ctx).getHttpSender();
         try (HttpSender.Response resp = httpSender.send(httpSender.get(GRADLE_VERSIONS_ALL_URL).build())) {
             if (resp.isSuccessful()) {
-                return new ObjectMapper()
+                List<GradleVersion> gradleVersions = new ObjectMapper()
                         .registerModule(new ParameterNamesModule())
                         .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
                         .readValue(resp.getBody(), new TypeReference<List<GradleVersion>>() {
                         });
+                List<GradleVersion> allGradleVersions = new ArrayList<>(gradleVersions.size() * 2);
+                for (GradleVersion gradleVersion : gradleVersions) {
+                    allGradleVersions.add(new GradleVersion(
+                            gradleVersion.version,
+                            gradleVersion.downloadUrl,
+                            DistributionType.Bin,
+                            gradleVersion.checksumUrl,
+                            gradleVersion.wrapperChecksumUrl
+                    ));
+                    allGradleVersions.add(new GradleVersion(
+                            gradleVersion.version,
+                            gradleVersion.downloadUrl.replace("-bin.zip", "-all.zip"),
+                            DistributionType.All,
+                            gradleVersion.checksumUrl == null ? null : gradleVersion.checksumUrl.replace("-bin.zip", "-all.zip"),
+                            gradleVersion.wrapperChecksumUrl
+                    ));
+                }
+                return allGradleVersions;
+            }
+            throw new IOException("Could not get Gradle versions. HTTP " + resp.getCode());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static List<GradleVersion> listAllPrivateArtifactoryVersions(String artifactoryUrl, ExecutionContext ctx) {
```

Comment:
I think it would make sense to create an integration test that works against Gradle Wrappers placed in our Artifactory instance.

Reply:
The two test cases -- `supportsSemverAgainstJFrogArtifactory` and `supportsSemverAgainstJFrogArtifactoryWithPath` -- should satisfy this plenty while being less prone to latency and systemic issues. The JSON content is identical in shape to what is returned from an Artifactory instance's REST API.